generic_config_updater/patch_sorter.py

```diff
@@ -0,0 +1,965 @@
+
+    JsonMove on the other hand allows that given the target_config_tokens i.e. the target_config path,
+    and current_config_tokens i.e. current_config path where the update needs to happen.
+    """
+    def __init__(self, diff, op_type, current_config_tokens, target_config_tokens=None):
+        operation = JsonMove._to_jsonpatch_operation(diff, op_type, current_config_tokens, target_config_tokens)
+        self.patch=jsonpatch.JsonPatch([operation])
+        self.op_type=operation[OperationWrapper.OP_KEYWORD]
+        self.path=operation[OperationWrapper.PATH_KEYWORD]
+        self.value=operation.get(OperationWrapper.VALUE_KEYWORD, None)
+
+        self.op_type=op_type
+        self.current_config_tokens=current_config_tokens
+        self.target_config_tokens=target_config_tokens
+
+    @staticmethod
+    def _to_jsonpatch_operation(diff, op_type, current_config_tokens, target_config_tokens):
+        operation_wrapper=OperationWrapper()
+        path_addressing=PathAddressing()
+
+        if op_type == OperationType.REMOVE:
+            path = path_addressing.create_path(current_config_tokens)
+            return operation_wrapper.create(op_type, path)
+
+        if op_type == OperationType.REPLACE:
+            path = path_addressing.create_path(current_config_tokens)
+            value = JsonMove._get_value(diff.target_config, target_config_tokens)
+            return operation_wrapper.create(op_type, path, value)
+
+        if op_type == OperationType.ADD:
+            return JsonMove._to_jsonpatch_add_operation(diff, current_config_tokens, target_config_tokens)
+
+        raise ValueError(f"OperationType {op_type} is not supported")
+
+    @staticmethod
+    def _get_value(config, tokens):
+        for token in tokens:
+            config=config[token]
+
+        return copy.deepcopy(config)
+
+    @staticmethod
+    def _to_jsonpatch_add_operation(diff, current_config_tokens, target_config_tokens):
+        """
+        Check description of JsonMove class first.
+
+        ADD operation path can refer to elements that do not exist, so to convert JsonMove to JsonPatch operation
+        We need to remove the non-existing tokens from the current_config path and move them to the value.
+
+        Example:
+        Assume Target Config:
+        {
+            "dict1":{
+                "key11": "value11"
+            }
+        }
+        Assume Current Config:
+        {
+        }
+        Assume JsonMove:
+        op_type=add, current_config_tokens=[dict1, key11], target_config_tokens=[dict1, key11]
+
+        Converting this to operation directly is incorrect since the current_config tokens contains 'dict1' which does
+        not exist in Current Config. Instead we convert to:
+        {"op":"add", "path":"/dict1", "value":{"key11": "value11"}}
+        """
+        operation_wrapper=OperationWrapper()
+        path_addressing=PathAddressing()
+
+        # if path refers to whole config, then just create operation
```

Comment:
improve comment by saying:
if path refers to whole config i.e. no tokens, then just create the operation